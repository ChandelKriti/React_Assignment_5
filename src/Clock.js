import React , { useState } from 'react'
import './Clock.css';

function Clock() {


    let time = new Date().toLocaleTimeString();

    const [clockTime, clockSetTime] = useState(time);

    const updateTime = () => {
        time = new Date().toLocaleTimeString();
        clockSetTime(time);
    }


    setInterval(updateTime,1000);

    let today= new Date();
    let todayDate = today.getDate() + '-' + ((today.getMonth() + 1)) + '-' + today.getFullYear() ;

    const [noToggle, setToggle] = useState('');

    const setDate = () =>{
       
        if(noToggle==='')
        {
            setToggle(todayDate);
        }
        else{
            setToggle('');
        }

    }

     

    return (
        <div>

        <div>
            <label className="switch">
                <input type="checkbox" onClick={setDate} />
                <span className="slider round"></span>

            </label>

            
        </div>
        
        <div >
            <h1><span className="center">{time}<br /><br />{noToggle}</span></h1>
            

        </div>
        

        </div>
    
    )
}

export default Clock;
