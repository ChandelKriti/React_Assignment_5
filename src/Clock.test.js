import React from 'react';
import { shallow, mount } from 'enzyme';
import Clock from './Clock';

describe('Clock',() => {

  const container = shallow(<Clock />)

  it ('should match snapshot', () => {
      expect(container.html()).toMatchSnapshot();
  });

  it('should have input field', () => {
      expect(container.find('input[type="checkbox"]').length).toEqual(1);
  })

})

describe('Clock', () => {
    let clock;
    let onClick;

    beforeEach(() => {
        onClick = jest.fn();
        clock = mount(<Clock onClick={onClick} />)
    })

    it('clock requires onClick prop', () => {
        expect(clock.props().onClick).toBeDefined()
    })
})
