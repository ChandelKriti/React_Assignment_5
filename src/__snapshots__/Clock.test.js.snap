// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock should match snapshot 1`] = `"<div><div><label class=\\"switch\\"><input type=\\"checkbox\\"/><span class=\\"slider round\\"></span></label></div><div><h1><span class=\\"center\\">2:49:38 PM<br/><br/></span></h1></div></div>"`;
